### ISSUE TITLE
Dynamic Recipients

### ISSUE THREAD

**Issue by mufi-siebird** [doublesecretagency/craft-notifier#23]
https://plugins.doublesecretagency.com/notifier/recipients/types/dynamic-recipients

Do we have any approx date on when we will be able to use this feature?

**Comment by lindseydiloreto** [comment]
Great question... I don't have a specific ETA, hopefully within the next month or so. We're currently focused on making sure all of our plugins are compatible and stable with Craft 5.

Out of curiosity, what is the specific use-case? How do you picture yourself using Dynamic Recipients?

I'm trying to get a sense of exactly what people will need out of that feature.

**Comment by johnj24** [comment]
Hi, Lindsey. In the beta version, I believe this was called Custom Users, correct? At that time, I was able to get a collection of users using some custom twig code in the control panel. In my instance, I used a Matrix field to let the client select which users to include on an entry and those selected users were the only ones who got the notifications when the entry was created/updated.

Is this how you see Dynamic Recipients working? Is there any way to do this kind of thing currently with version 1 or 2? There doesn't seem to be a way from what I can tell.

Also, while I'm here, a big thanks for the revamped UI (title/description fields especially) and the new Assets upload trigger.

**Comment by lindseydiloreto** [comment]
Hi John! That's correct, Dynamic Recipients are an evolution of Custom Users.

Technically, almost any workaround is possible using the [`{% skipMessage %}`](https://plugins.doublesecretagency.com/notifier/messages/skip) tag. In Twig, you can write the logic which permits or denies each individual message from being sent.

For a more complex situation like yours, the message template might include something like this...

```twig
{##
 # Use `entry.yourMatrixField` to compile an array of permitted User IDs.
 # The `recipientIds` is a hard-coded example of what to provide.
 #}

{# Example array #}
{% set recipientIds = [33, 42, 101] %}

{# If not one of the intended recipients, skip sending this message #}
{% if recipient.user.id not in recipientIds %}
    {% skipMessage "User is not an intended recipient." %}
{% endif %}
```

Worth noting that you'll need to cast a wider net when specifying which Users should receive the notification... the `skipMessage` tag filters _out_ recipients from a larger pool.

If you know the subset of Users which could potentially receive the message, you can select just those [Users](https://plugins.doublesecretagency.com/notifier/recipients/types/selected-users) (or a [group](https://plugins.doublesecretagency.com/notifier/recipients/types/selected-groups)) in the Notification configuration.

---

If all of that sounds like too much effort, you could theoretically sit on the beta version a little longer. I don't know the exact ETA, but I'm hopeful that it will be out within the next 30 days.

**Comment by steverowling** [comment]
Are you planning on adding the Dynamic Recipients feature to the Craft 4 version of the plugin? I have a potential client that has a requirement for notifications, but would need the dynamic recipients feature to make it work for them. For various reasons, the site can't be upgraded to Craft 5 for a few months, but they need the notification requirement before then.

**Comment by lindseydiloreto** [comment]
@steverowling Yes, that is still very much the plan. Sorry for blowing way past the previous ETA.

We've been knocked a little off course by a backlog of client work (and some expected summer travels). Notifier 1.1 is sitting in a half-finished state, and I'm eager to get back and button it up. I'm hesitant to provide another ETA, but I promise this feature is near the top of the list.

Fingers crossed I'll be turning my attention back to Notifier within the next few weeks. 🤞

**Comment by theurbanu** [comment]
Hello, just checking in to see if there's any update on this feature being added?

**Comment by lindseydiloreto** [comment]
As much as I hate to admit it, this feature is stalled out at the moment. [Last month](https://www.doublesecretagency.com/blog/notifier-plugin-v1-1-v2-1) we were able to wrap up some loose ends on Notifier, but unfortunately Dynamic Recipients got put back on the shelf until we're able to tackle another round of updates.

Truly sorry for the lengthy delay and repeated missed estimates! 🙏 

If anyone is interested in sponsoring this feature, that would help greatly in moving it up the priority list. Without sponsorship, it's unlikely we'll get to this before late 2025.

**Comment by BrandKits** [comment]
Hi Lindsey, just following up to see if you've made any progress with this feature? We're testing it for a new client, but really need dynamic recipients for it to work (use case is a Jobs site with employers and candidates, users and entries requiring notifications). Many thanks.

**Comment by lindseydiloreto** [comment]
Thanks for checking in.

It's still stuck near the top of the list. I agree with everyone that this is a hugely important issue, and a blocker for several of you.

We're going to be investing a lot of time into plugin development in early 2026. Notifier will be a high priority item, and this ticket is probably first on the list. Without making any explicit promises, I have high hopes for Q1 2026.

Will try to keep everyone posted here with any updates. 👍

**Comment by lindseydiloreto** [comment]
Excellent news, **Dynamic Recipients** has been officially released in [v3.0.0](https://github.com/doublesecretagency/craft-notifier/releases/tag/3.0.0). 🙏 🎈 

Sorry for the extremely long wait, and the bait-and-switch if you were a beta user. It was a bit of a roller coaster, but I think the final version of [Dynamic Recipients](https://plugins.doublesecretagency.com/notifier/recipients/types/dynamic-recipients) turned out great!
